### PR TITLE
Feat/improved readme

### LINKED ISSUE #21: Add usage/configuration to readme
## Issue

Currently the readme is pretty empty, this needs to be changed if we want someone to actually use it.

### Problem to solve

Write configuration and how to use the downscaler in the readme, do not just copy form pykubedownscaler (please!)

### Further details

Most configuration can be found in Obsidian Notes and just has to be restructured and made more clear to users.

### Proposal

Add these to the Readme:
- Use cases
- Concepts (esp. for Layer system)
- Configuration
- How to switch from py-kube-downscaler
- Differences to py-kube-downscaler

Add readme buttons like in [caas/py-kube-downscaler/README.md](https://github.com/caas-team/py-kube-downscaler/blob/5dac93592f22ead37b5f581fd2db1b76c47bc41d/README.md#L3-L6)

### Who can address the issue

Anyone, preferably with access to Obsidian Notes, if not just ask for the md file by @-ing me

### PR DESCRIPTION
**Pull request by jonathan-mayer**
## Motivation

Closes #21 

As it is pretty important that anyone can understand this documentation this pr should be reviewed from as many people as possible, so at least both codeowners.

## Changes

- Add an actual readme
- Add some more pre-commit-checks
- Fix some things pre-commit whined about
- Make Force-* more consistent with other boolean values

## Tests done

- [X] unit tests passed

## TODO

- [x] I've assigned myself to this PR

### CODE REVIEW

**Review by jon4skl — APPROVED** [review]
looks good to me

**Comment by jonathan-mayer** [comment]
I will add the changes made in #31 here before it gets merged.

**Comment by jonathan-mayer** [comment]
ready to merge once approved!

File: README.md
Comment (JTaeuber) [inline README.md:56]: Both have to be daemonset

**Review by JTaeuber — CHANGES_REQUESTED** [review]

File: README.md
Comment (JTaeuber) [inline README.md:147]: This almost makes it sound as if the field is left empty it allows all kinds of downscaling. I would rephrase to something like `Enables downscaling for the specified resources` or something similar

File: README.md
Comment (JTaeuber) [inline README.md:151]: I'd be more specific here and say that it excludes the specified namespaces from the downscale process

File: README.md
Comment (JTaeuber) [inline README.md:152]: default from values.yaml of the helm chart:
```yaml
excludedNamespaces:
  - go-kube-downscaler
  - kube-downscaler
  - kube-system
```

File: README.md
Comment (JTaeuber) [inline README.md:157]: I don't fully understand. Is this also a command line argument? Two dashes are missing if so.

File: README.md
Comment (JTaeuber) [inline README.md:182]: Dead link since there is no '--exclude-deployments' argument.

File: CONTRIBUTING.md
Comment (JTaeuber) [inline CONTRIBUTING.md:41]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:199]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:207]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:246]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:434]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:436]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:442]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:478]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:474]: Please use either eg or e.g.

File: README.md
Comment (JTaeuber) [inline README.md:382]: Has to be upscale-period

File: README.md
Comment (JTaeuber) [reply README.md:434]: It also is an uptime annotation.

File: README.md
Comment (jonathan-mayer) [reply README.md:152]: this is documentation on the cli arguments and their defaults, in the code the defaults are:
```
{"kube-system", "kube-downscaler"}
```
either we leave it like this, change the defaults in the code, change the defaults in the helm chart or change the helm chart to by default use the defaults by the code by not overwriting them even if they aren't set

Comment (jonathan-mayer) [reply README.md:152]: changed the defaults in the values.yaml, but we should look into implementing the last option i mentioned. basically just wrapping the line that sets the env variable in an if statement that checks if the excludeNamespaces var is set

**Review by jon4skl — APPROVED** [review]

**Review by JTaeuber — APPROVED** [review]
LGTM

**Pull request merged**
